registry/client/repository_test.go

```diff
@@ -1019,12 +1029,12 @@ func TestManifestTagsPaginated(t *testing.T) {
 			"Last-Modified":  {time.Now().Add(-1 * time.Second).Format(time.ANSIC)},
 		})
 		if i < 2 {
-			headers.Set("Link", "<"+s.URL+"/v2/"+repo.Name()+"/tags/list?n=1&last="+tagsList[i]+`>; rel="next"`)
+			headers.Set("Link", "<"+s.URL+"/v2/"+repo.Name()+"/tags/"+ref+"?n=1&last="+tagsList[i]+`>; rel="next"`)
```

Comment:
Let's try and setup tests that have the following:

1. Older client hits `tags/list` and is redirected.
2. Newer client hits `tags/_list` against old registry and falls back.
3. Newer client hits `tags/_list` against new registry.